src/puppetlabs/ezbake/core.clj

```diff
@@ -699,6 +705,40 @@ Additional uberjar dependencies:
 (defmulti action
   (fn [action & args] action))
 
+(defn munge-deps
+  ;; Takes a map generated by deputils/generate-dependency-map and transforms
+  ;; it into a map where the keys are the dependencies themselves with sub
+  ;; keys of versions and sub-dependencies.
+  ;;
+  ;; Example input being:
+  ;; {[project-name project-version extra-info] {[dependency-name dependency-version] {}}}
+  ;; Which would be transformed into the following map:
+  ;; {project-name {"version" project-version "dependencies" {dependency-name {"version" dependency-version}}}}
+  [deps]
+  (into {} (concat (for [[dep dependencies] deps
+                         :let [[project-name version] dep
+                               munged-dependencies (munge-deps dependencies)
+                               metadata (if (empty? munged-dependencies)
+                                          {"version" version}
+                                          {"version" version "dependencies" munged-dependencies})]]
+                         {(str project-name) metadata}))))
+
+(defn generate-build-metadata-files
+  [lein-project]
+  (let [dep-map (deputils/generate-dependency-map lein-project)
+        deps (munge-deps dep-map)
+        metadata {"packaging_type" {"ezbake" (get-ezbake-version lein-project)}
+                  "git_sha" (get-git-sha)
+                  "build_time" (local-time/format-local-time (local-time/local-now) :rfc822)
```

Comment:
I think it's worth spending a few minutes in Ruby to try and actually see how we're going to parse and use this date value.

The format that this generates: `"Thu, 29 Jun 2017 12:03:35 -0700"`